spring-kafka/src/main/java/org/springframework/kafka/config/KafkaStreamsTopologyCustomizer.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.config;
+
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.Topology;
+
+/**
+ * Callback interface that can be used to configure {@link KafkaStreams} directly.
+ *
+ * @author Renato Mefi
+ *
+ * @since 2.4.0
+ *
+ * @see StreamsBuilderFactoryBean
+ */
+@FunctionalInterface
+public interface KafkaStreamsTopologyCustomizer {
```

Comment:
Why can't we leave just with the `java.util.function.Consumer`?
Even for composition we can use its `Consumer<T> andThen(Consumer<? super T> after)`.

Reply:
Thanks for the info!

Given that, I followed the same interface as `KafkaStreamsCustomizer` for user experience, should I still try with the `java.util.function.Consumer`?

Given the added info by @garyrussell I think we have 2 paths: Follow with the current inferface or breaking all the customizers with your suggestion!

Which path should I take?

Reply:
I think the `KafkaStreamsCustomizer` abstraction has been chosen for Spring Boot auto-configuration  purposes.
The same might happened with this option as well.
But I think it won't hurt to stick with the particular generic for the `Consumer`.
I even don't think that this customization option should be considered as a bean in the target application configuration: a lambda for consumer could do the trick with access to any possible bean in the configuration.

If we don't go a `java.util.function.Consumer` approach, I don't think we need that `CompositeKafkaStreamsTopologyCustomizer`: the same `andThen()` implementation really will lead us to the same result. Even if we will end up with a bean, which could be useful in Spring Boot auto-configuration.

Reply:
I don't see any problems adding this to 2.3; I generally concur with Artem but, since we already have a precedent of an explicit interface, I guess it would be cleaner to stick with it here too. I don't see any reason to break the API.

Reply:
Good. Let's stick with an explicit functional interface then!
But let's try to implement that `andThen()` instead of composite! 😉